collect_app/src/main/java/org/odk/collect/android/configure/SettingsImporter.kt

```diff
@@ -0,0 +1,79 @@
+package org.odk.collect.android.configure
+
+import org.json.JSONException
+import org.json.JSONObject
+import org.odk.collect.android.application.initialization.SettingsPreferenceMigrator
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.shared.Settings
+
+class SettingsImporter(
+    private val settingsProvider: SettingsProvider,
+    private val preferenceMigrator: SettingsPreferenceMigrator,
+    private val settingsValidator: SettingsValidator,
+    private val generalDefaults: Map<String, Any>,
+    private val adminDefaults: Map<String, Any>,
+    private val settingsChangedHandler: SettingsChangeHandler
+) {
+
+    @JvmOverloads
+    fun fromJSON(json: String, uuid: String? = null): Boolean {
```

Comment:
I think we should use `projectId` here rather than `uuid` so we know what the ID refers to. I've been think it also makes sense to rename the field on project, but holding off on doing that. The fact that the ID is a UUID isn't really as important outside the repo itself.

Reply:
Fixed.